### DISCUSSION TITLE
Resolution Center doesn't work

### DISCUSSION THREAD

**Discussion by jannyss** [fastlane/fastlane#17220] [Q&A] [answered]
Try to get information from Resolution Center using this code:

```
Spaceship::Tunes.login(ARGV[0], ARGV[1])
puts Spaceship::Tunes.client.select_team
app = Spaceship::Tunes::Application.find(ARGV[3])

if app == nil
  puts "Application #{ARGV[3]} not found!"
  Process.exit(1)
end

print_build_info('live', app.live_version)
print_build_info('edit', app.edit_version)

if app.resolution_center != nil
  print("resolution_center:" + JSON.unparse(app.resolution_center))
end
```

And got this error:
`2.159.0/spaceship/lib/spaceship/client.rb:719:in` `parse_response': {"data"=>nil, "messages"=>{"warn"=>nil, "info"=>nil, "error"=>["Not Found"]}, "statusCode"=>"ERROR"} (Spaceship::UnexpectedResponse)`

I've got this error since September 4th.

Please, advise me, what should I do fix in my code.
Used the latest version of fastlane (2.159.0)

**Comment by hnakamura-fsv** [comment, marked as answer]
We've had the same issues - the issue is to do with getting the old `Spaceship::Tunes::Application` object and the related live/edit versions.

You'll need to replace all of your calls to `Spaceship::Tunes::Application` with `Spaceship::ConnectAPI::App`, and replace any calls to live_version and edit_version to the corresponding calls under the new ConnectAPI classes. e.g: https://github.com/fastlane/fastlane/blob/f2cbf81619b722af5a549fd059f2678a9d471880/spaceship/lib/spaceship/connect_api/models/app.rb#L200

For resolution center, one thing to try is instead of retrieving it through the `Spaceship::Tunes::Application` object, you can try to call `Spaceship::Tunes.client.get_resolution_center(app_id, platform)` and it might work (but it seems a bit flakey for me)

**Reply by jannyss** [reply]
Thank you a lot!

**Comment by max-ott** [comment]
Check out https://github.com/fastlane/fastlane/issues/17298 for investigation with regards to `ConnectAPI`.
